lib/appsignal/probes/erlang_probe.ex

```diff
@@ -10,31 +10,45 @@ defmodule Appsignal.Probes.ErlangProbe do
 
   defp io_metrics do
     {{:input, input}, {:output, output}} = :erlang.statistics(:io)
-    @appsignal.set_gauge("erlang_io", Kernel.div(input, 1024), %{type: "input"})
-    @appsignal.set_gauge("erlang_io", Kernel.div(output, 1024), %{type: "output"})
+    set_gauge("erlang_io", Kernel.div(input, 1024), %{type: "input"})
+    set_gauge("erlang_io", Kernel.div(output, 1024), %{type: "output"})
   end
 
   defp scheduler_metrics do
-    @appsignal.set_gauge("erlang_schedulers", :erlang.system_info(:schedulers), %{type: "total"})
+    set_gauge("erlang_schedulers", :erlang.system_info(:schedulers), %{type: "total"})
 
-    @appsignal.set_gauge(
+    set_gauge(
       "erlang_schedulers",
       :erlang.system_info(:schedulers_online),
       %{type: "online"}
     )
   end
 
   defp process_metrics do
-    @appsignal.set_gauge("erlang_processes", :erlang.system_info(:process_limit), %{type: "limit"})
+    set_gauge("erlang_processes", :erlang.system_info(:process_limit), %{type: "limit"})
 
-    @appsignal.set_gauge("erlang_processes", :erlang.system_info(:process_count), %{type: "count"})
+    set_gauge("erlang_processes", :erlang.system_info(:process_count), %{type: "count"})
   end
 
   defp memory_metrics do
     memory = :erlang.memory()
 
     Enum.each(memory, fn {key, value} ->
-      @appsignal.set_gauge("erlang_memory", Kernel.div(value, 1024), %{type: to_string(key)})
+      set_gauge("erlang_memory", Kernel.div(value, 1024), %{type: to_string(key)})
     end)
   end
+
+  defp set_gauge(name, value, tags) do
+    @appsignal.set_gauge(
+      name,
+      value,
+      Map.merge(tags, %{host_metric: "", hostname: hostname()})
+    )
+  end
+
+  defp hostname do
+    {:ok, hostname} = :inet.gethostname()
+    config = Application.get_env(:appsignal, :config)
+    config[:hostname] || hostname
```

Comment:
We're asking for `:inet.gethostname/0` now, even if we have a configured hostname. Let's check that config value first, which would look something like this (beware, untested):

``` elixir
case Application.fetch_env!(:appsignal, :config)[:hostname] do
  nil -> 
    {:ok, hostname} = :inets.gethostname()
    hostname
  hostname -> hostname
end
```